Tasks/AppStoreRelease/app-store-release.ts

```diff
@@ -270,39 +272,51 @@ async function run() {
             // See https://github.com/fastlane/fastlane/blob/master/pilot/lib/pilot/options.rb for more information on these arguments
             let pilotCommand: ToolRunner = tl.tool('fastlane');
             let bundleIdentifier: string = tl.getInput('appIdentifier', false);
-            if (isUsingApiKey) {
-                pilotCommand.arg(['pilot', 'upload', '--api_key_path', apiKeyFilePath, '-i', filePath]);
+            let externalTestersGroups: string = tl.getInput('externalTestersGroups');
+
+            if (distributeOnly) {
+                if (isUsingApiKey) {
+                    pilotCommand.arg(['pilot', 'distribute', '--api_key_path', apiKeyFilePath]);
+                } else {
+                    pilotCommand.arg(['pilot', 'distribute', '-u', credentials.username]);
+                }
+                pilotCommand.argIf(appBuildNumber, ['--build_number', appBuildNumber]);
+                pilotCommand.argIf(bundleIdentifier, ['-a', bundleIdentifier]);
+                pilotCommand.argIf(externalTestersGroups, ['--groups', externalTestersGroups]);
             } else {
-                pilotCommand.arg(['pilot', 'upload', '-u', credentials.username, '-i', filePath]);
-            }
-            let usingReleaseNotes: boolean = isValidFilePath(releaseNotes);
-            if (usingReleaseNotes) {
-                if (!credentials.fastlaneSession) {
-                    tl.warning(tl.loc('ReleaseNotesRequiresFastlaneSession'));
+                if (isUsingApiKey) {
+                    pilotCommand.arg(['pilot', 'upload', '--api_key_path', apiKeyFilePath, '-i', filePath]);
```

Comment:
Nit: could you please move it to separate method?

Reply:
I believe that now we do not need to do this. Probably we should refactor the entire task later in separate PR.